### ISSUE TITLE
password not saving correctly (reproducible)

### ISSUE THREAD

**Issue by githubismyresume** [spesmilo/electrum#6071]
MacOS Desktop client:

steps to reproduce

-create any password with alphabetic words seperated by special characters: !,$,%,* (only tested these specific inputs) 

-also include 1 or more numeric characters somewhere in password.

EX:

Password!password4!password

Current behaviour (bug)

after creating password successfully, electrum will fail to validate upon restart of the application. 
reproduced half a dozen times locally 

/label ~Bug

**Comment by SomberNight** [comment]
Are you trying to create a new wallet, or change the password of an existing one?

**Comment by SomberNight** [comment]
Is "Encrypt wallet file" enabled? I assume it is as that is the default.
Are you running the 3.3.8 official release (.dmg)? (as opposed to e.g. from source)

I have now tested on MacOS 10.14 with the 3.3.8, setting specifically your example password, but it works for me.

**Comment by githubismyresume** [comment]
@SomberNight 


> Are you trying to create a new wallet, or change the password of an existing one?

yes, new 

> Is "Encrypt wallet file" enabled? I assume it is as that is the default.

yes

>  Are you running the 3.3.8 official release (.dmg)? (as opposed to e.g. from source)

i installed from brew cask (release 3.3.8)

I'm on Catalina at the moment. I'm glad to hear it worked for you: it points to the problem being somewhere on my system, I suppose. 

Are there any listed password restrictions? I couldn't find any with a quick check last night.

**Comment by SomberNight** [comment]
> Are there any listed password restrictions? I couldn't find any with a quick check last night.

No, there are no explicit restrictions. The password is hashed so the application really doesn't care.

Could this be related? https://github.com/spesmilo/electrum/issues/5038
(and upstream https://bugreports.qt.io/browse/QTBUG-73363)

**Comment by githubismyresume** [comment]
Great catch. That certainly seems like it could be the cause. Should I close this bug report?

**Comment by SomberNight** [comment]
Can you reproduce this issue with only ASCII characters? If so, it might be a different issue.

Maybe we should add an option to "unmask" the characters in these TextEdits.

**Comment by githubismyresume** [comment]
Just reproduced the following:

1.  created new wallet
2. generated random password (ASCII characters only) with Keychain 
3. copy/pasted password from Keychain GUI to electron GUI 
4. saved wallet, exited and restarted electron

5. typed password manually, failed.
6. pasted password (same as step 3), successfully logged in.

oddly enough, typing the password, then exporting the masked text shows that it's identical to the Keychain paste.

**Comment by ecdsa** [comment]
I cannot reproduce this bug (on linux).
@githubismyresume is that with Electrum or Electron-cash ?
